cpp/include/cuvs/neighbors/cagra.hpp

```diff
@@ -107,6 +114,52 @@ struct index_params : cuvs::neighbors::index_params {
   bool attach_dataset_on_build = true;
 };
 
+/**
+ * @brief Create a CAGRA index parameters compatible with HNSW index
+ *
+ * @param dataset The shape of the input dataset.
+ * @param M HNSW index parameter M.
+ * @param ef_construction HNSW index parameter ef_construction.
+ * @param metric The distance metric to search.
+ *
+ *
+ * * IMPORTANT NOTE *
+ *
+ * The reference HNSW index and the corresponding from-CAGRA generated HNSW index will NOT produce
+ * the same recalls and QPS for the same parameter `ef`. The graphs are different internally. For
+ * the same `ef`, the from-CAGRA index likely has a slightly higher recall and slightly lower QPS.
+ * However, the Recall-QPS curves should be similar (i.e. the points are just shifted along the
+ * curve).
+ *
+ * Usage example:
+ * @code{.cpp}
+ *   using namespace cuvs::neighbors;
+ *   raft::resources res;
+ *   auto dataset = raft::make_device_matrix<float, int64_t>(res, N, D);
+ *   auto cagra_params = hnsw_to_cagra_params(dataset.extents(), M, efc);
+ *   auto cagra_index = cagra::build(res, cagra_params, dataset);
+ *   auto hnsw_index = hnsw::from_cagra(res, hnsw_params, cagra_index);
+ * @endcode
+ */
+template <typename T, typename IdxT>
+auto hnsw_to_cagra_params(raft::matrix_extent<int64_t> dataset,
```

Comment:
Please remove the unused template parameters.
The function will become a non-template and you'll need to decide in which compilation unit to place it. If #1448 doesn't get merged before this PR, I'd suggest to just copy the relevant bits here (create a separate `cagra.cpp` unit).